Plotting functionality to integrate ClinVar and AlphaMissense data

tram-nguyen-n opened the pull request:
This PR adds plotting functionality to integrate ClinVar and AlphaMissense data. The function uses functionality already available in the AlphaMissenseR package to access AlphaMissense information. New components include:

1. `plot_clinvar()` in `plot_clinvar.R`
2. Test cases in `test-plot_clinvar.R`
3. Within `data/`, processed ClinVar data derived from the supplemental data included in the AlphaMissense publication available as`clinvar_data.rda`
4. Within `inst/scripts`, code to produce the `clinvar_data.rda` from the raw data in `make_clinvar_dataset.R`
5. Within the `inst/extdata`, I also included the ClinVar raw data from the paper as `science.adg7492_data_s5.csv.gz`
6. `clinvar_data.R`to provide documentation for the ClinVar dataset that has been added
7. Brief vignette called `d_integrate_ClinVar.Rmd`

I updated the DESCRIPTION file to include a version bump to 1.1.3, added R (>= 4.3.0) to Depends section given the new .rda, and included `LazyData: true` to deal with a `CMD Check` warning  that “Variables with usage in Rd file 'clinvar_data.Rd' but not in code: ‘clinvar_data’”.

I ran `BiocCheck()` and `R CMD Check` and they passed successfully. The code coverage was also at a 100% for the `plot_clinvar()` function. 

This is our first pass at incorporating the ClinVar and AlphaMissense data for benchmarking and visualization. We’re very happy to have your feedback and thoughts! I am can discuss/work further either on here or we can also set up a video call. Thanks very much in advance, @mtmorgan!

mtmorgan commented on NAMESPACE:
```diff
@@ -15,7 +15,10 @@ export(db_disconnect_all)
 export(db_range_join)
 export(db_tables)
 export(db_temporary_table)
+export(plot_clinvar)
```
I'd prefer `clinvar_plot`, which follows the prefix scheme `af_`, `am_`, etc., used in the package.

mtmorgan commented on NAMESPACE:
```diff
@@ -15,7 +15,10 @@ export(db_disconnect_all)
 export(db_temporary_table)
+export(plot_clinvar)
 export(to_GPos)
+import(dplyr)
```
please enumerate (`@importFrom`) specific functionality for dplyr and ggplot2

mtmorgan commented on R/clinvar_data.R:
```diff
@@ -0,0 +1,17 @@
+#' Default ClinVar Data
```
I think this and the plotting functions should be in a single file ClinVar.R with `@rdname` used so that a single 'ClinVar.Rd' help page is created.

Can you be more precise in the 'Derived from' description, mentioning the specific table and indicating that the script used to process the data is available at `system.file(package = "AlphaMissenseR", "scripts", "clinvar_make_dataset.R")`

mtmorgan commented on inst/scripts/make_clinvar_dataset.R:
```diff
@@ -0,0 +1,34 @@
+## code to prepare `clinvar_data.rda` dataset
+
+#wget https://www.science.org/doi/suppl/10.1126/science.adg7492/suppl_file/science.adg7492_data_s1_to_s9.zip
```
Use `download.file()` / `unzip()` instead for a pure R solution

mtmorgan commented on R/clinvar_data.R:
```diff
@@ -0,0 +1,17 @@
+#' Derived from the supplemental table of the AlphaMissense
+#'      [\[2023\]](https://www.science.org/doi/10.1126/science.adg7492) paper.
+#'
+#' @format dataframe with 82872 rows and 5 variables:
```
actually a 'tbl' ?

mtmorgan commented on R/clinvar_data.R:
```diff
@@ -0,0 +1,17 @@
+#'      \item{uniprot_id}{UniProt accession identifier.}
+#'      \item{transcript_id}{Ensembl transcript identifier.}
+#'      \item{protein_variant}{Protein variant identifier.}
+#'      \item{cv_class}{Binary ClinVar class. 0 for benign or 1 for pathogenic.}
```
Since the data has been processed already, maybe it makes better sense for this to be a factor with levels 'benign' and 'pathogenic'

mtmorgan commented on R/plot_clinvar.R:
```diff
@@ -0,0 +1,319 @@
+    }
+
+    ## Take alphamissense_table and filter for the uniprotId
+    alphamissense_table <- am_table |>
```
I like to have the variable assignment separate from the stanza describing the transformation
```
alphamissense_table <-
    am_table |>
    ...
```

mtmorgan commented on R/plot_clinvar.R:
```diff
@@ -0,0 +1,319 @@
+    ## Check if table is empty after filtering
+    ## This will work for a tibble or a data.frame
+    if (!nrow(alphamissense_table)) {
+        stop("No AlphaMissense information found for the protein ",
```
Because this is displayed as `Error: ...`, it makes sense to *not* capitalize the sentence. My convention in the package seems to not include terminating punctuation. And whenever I need to extend arguments over more than one line I like to put all arguments in separate lines, indented by 4 spaces. Can you provide the user with whatever their input was, with single quotes surrounding so that leading / trailing white space is visible?
```
    if (!nrow(alphamissense_table)) {
        stop(
            "no AlphaMissense information found for the protein ",
            "accession '", uID, "'; check that the UniProt ID is correct"
        )
    }
```

mtmorgan commented on R/plot_clinvar.R:
```diff
@@ -0,0 +1,319 @@
+    function(cv_table, uID)
+{
+    if (missing(cv_table)) {
+        message("'clinvar_table' not provided, using default ",
```
In this package I used `spdl::info()` and `spld::warn()` instead of message / warning, something like
```
        spdl::info(paste0(
            "'clinvar_table' not provided, using default ",
            "ClinVar dataset in AlphaMissenseR package"
        ))

mtmorgan commented on R/plot_clinvar.R:
```diff
@@ -0,0 +1,319 @@
+#' Filter the AlphaMissense table with uniprotID
+#'
+#' @noRd
+filter_am_table <-
```
follow function naming convention `clinvar_filter_am_table` throughout

mtmorgan commented on R/plot_clinvar.R:
```diff
@@ -0,0 +1,319 @@
+#' Filter the clinvar table with uniprot ID
+#'
+#' @noRd
+filter_cv_table <-
```
`clinvar_filter_cv_table <-` etc.

mtmorgan commented on R/plot_clinvar.R:
```diff
@@ -0,0 +1,319 @@
+        message("'clinvar_table' not provided, using default ",
+                "ClinVar dataset in AlphaMissenseR package")
+
+        data_env <- new.env(parent = emptyenv())
```
If we make `clinvar_data` a table in the database, then this section will be moot. But I think that since you've used `LazyLoad: true` in the DESCRIPTION, the variable `clinvar_data` is just magically there, no need to use `data()`.

mtmorgan commented on R/plot_clinvar.R:
```diff
@@ -0,0 +1,319 @@
+    ## grab amino acid positions
+    am_table <- am_table |>
+        mutate(
+            aa_pos = as.numeric(
```
Is this actually integer-valued, `as.integer()`?

mtmorgan commented on R/plot_clinvar.R:
```diff
@@ -0,0 +1,319 @@
+        arrange(.data$code_color)
+
+    ## Grab the thresholds for AM pathogenicity to plot
+    combined_data |>
```
```
    combined_data |>
        group_by(.data$am_class) |>
        mutate(
            max = max(.data$am_pathogenicity, na.rm=TRUE),
            min = min(.data$am_pathogenicity, na.rm=TRUE)
        ) |>
        ungroup()

mtmorgan commented on R/plot_clinvar.R:
```diff
@@ -0,0 +1,319 @@
+    ## Create named vectors for all scale layers
+    colScale <- scale_colour_manual(
+        name = "code_color",
+        values = c("AM ambiguous" = "gray",
```
```
        values = c(
            "AM ambiguous" = "gray",
            "AM benign" = "#89d5f5",
             ...
         ))
```
etc

mtmorgan commented on R/plot_clinvar.R:
```diff
@@ -0,0 +1,319 @@
+                    "CV benign" = 1.5,
+                    "CV pathogenic" = 1.5))
+
+    cv_plot <- combined_table |>
```
```
    cv_plot <-
        combined_table |>
        ggplot(aes(.data$aa_pos, .data$am_pathogenicity)) +
        geom_point(aes(
            shape = .data$code_color,
            color = .data$code_color,
            ...
        )) +
   ...
```

mtmorgan commented on R/plot_clinvar.R:
```diff
@@ -0,0 +1,319 @@
+    function(uniprotId, alphamissense_table, clinvar_table)
+{
+    ## Validate arguments
+    stopifnot(isCharacter(uniprotId))
```
Can all the arguments be validated here, along the lines of
```
    stopifnot(
        isCharacter(uniprotId),
        is.data.frame(alphamisense_table),
        all(c("...", "..."), %in% colnames(alphamisense_table)),
        ...
    )
```

mtmorgan commented on inst/scripts/make_clinvar_dataset.R:
```diff
@@ -0,0 +1,34 @@
+           cv_class = label) |>
+    relocate('transcript_id', .after = 'uniprot_id')
+
+usethis::use_data(clinvar_data, overwrite = TRUE)
```
Maybe this isn't the right command with lazy data?

mtmorgan reviewed (commented):
Thanks Tram for the pull request! It looks great. There are a number of small formatting / etc suggestions throughout. Probably my comments are full of mistakes, so if it looks wrong let me know or do the thing I was thinking of.

My main comment is that I think the data should be put as a table in the database, rather than as a `data.frame`. The below is a diff that describes changes that might make this work, with the user saying `clinvar_data()` as a function call. The first time the data is loaded from the file inst/extdata/science.adg7492_data_s5.csv.gz and then the (duckdb) tbl returned from the database; subsequently it just gets the table.
```{diff}
diff --git a/R/AlphaMissense.R b/R/AlphaMissense.R
index 55707b2..35c70f7 100644
--- a/R/AlphaMissense.R
+++ b/R/AlphaMissense.R
@@ -150,7 +150,7 @@ am_available <-
 }
 
 am_data_import_csv <-
-    function(record, bfc, db_tbl_name, rname, fpath)
+    function(record, bfc, db_tbl_name, rname = NULL, fpath, delim = "\\t")
 {
     ## must be 'read_only = FALSE' so new database can be created. use
     ## 'managed = FALSE' so connection is independent of other
@@ -159,10 +159,13 @@ am_data_import_csv <-
     renew <- FALSE
     if (!db_tbl_name %in% db_tables(db_rw)) {
         spdl::info("downloading or finding local file")
-        file_path <- bfcrpath(bfc, rnames  = rname, fpath = fpath)
+        file_path <- fpath
+        if (!is.null(rname))
+            file_path <- bfcrpath(bfc, rnames  = rname, fpath = fpath)
         spdl::info("creating database table '{}'", db_tbl_name)
         sql <- sql_template(
-            "import_csv", db_tbl_name = db_tbl_name, file_path = file_path
+            "import_csv",
+            db_tbl_name = db_tbl_name, file_path = file_path, delim = delim
         )
         dbExecute(db_rw, sql)
 
@@ -278,9 +281,10 @@ am_data <-
 
     db_rname <- paste0("AlphaMissense_", record)
     db_tbl_name <- key$key
-    if (!NROW(bfcquery(bfc, db_rname)))
-        ## create the BiocFileCache record
+    if (!NROW(bfcquery(bfc, db_rname))) {
+        spdl::info("creating AlphaMissense database for record '{}'", record)
         bfcnew(bfc, db_rname)
+    }
     tbl <- am_data_import_csv(record, bfc, db_tbl_name, key$filename, key$link)
 
     switch(
diff --git a/R/clinvar_data.R b/R/clinvar_data.R
index c08a071..8378066 100644
--- a/R/clinvar_data.R
+++ b/R/clinvar_data.R
@@ -5,13 +5,25 @@
 #'
 #' @format dataframe with 82872 rows and 5 variables:
 #' \describe{
-#'      \item{cv_variant_id}{ClinVar variant identifer.}
-#'      \item{uniprot_id}{UniProt accession identifier.}
-#'      \item{transcript_id}{Ensembl transcript identifier.}
-#'      \item{protein_variant}{Protein variant identifier.}
-#'      \item{cv_class}{Binary ClinVar class. 0 for benign or 1 for pathogenic.}
-#'      }
+#'     \item{cv_variant_id}{ClinVar variant identifer.}
+#'     \item{uniprot_id}{UniProt accession identifier.}
+#'     \item{transcript_id}{Ensembl transcript identifier.}
+#'     \item{protein_variant}{Protein variant identifier.}
+#'     \item{cv_class}{Binary ClinVar class. 0 for benign or 1 for pathogenic.}
+#' }
 #'
-#' @examples
-#' data(clinvar_data)
-"clinvar_data"
+#' @export
+clinvar_data <-
+    function(record = ALPHAMISSENSE_RECORD, bfc = BiocFileCache())
+{
+    db_rname <- paste0("AlphaMissense_", record)
+    db_tbl_name <- "clinvar"
+    if (!NROW(bfcquery(bfc, db_rname))) {
+        spdl::info("creating AlphaMissense database for record '{}'", record)
+        bfcnew(bfc, db_rname)
+    }
+    fpath <- system.file(
+        package = "AlphaMissenseR", "extdata", "science.adg7492_data_s5.csv.gz"
+    )
+    am_data_import_csv(record, bfc, db_tbl_name, fpath = fpath, delim = ",")
+}
```
Probably there is some clever way for you to give me access to your repository so I could make these changes, if that is more convenient.

I'm not sure how this would influence the plotting function, it might be necessary to `dbplyr::collect()` the result of the duck-db operation. Also there's a tension around how much the data is processed before being inserted in the data base, e.g., when is the id column split? In the end I think the answer is 'give the data to the user the same way you get it', so that for instance if you've been doing your own work with the data outside of the package, there aren't any obstacles to starting to use the package. Also there's no way for you to make a mistake -- it's always the upstream person's fault!

tram-nguyen-n commented:
Thanks so much for your feedback! I'll start incorporating your suggestions. I've also added you as a collaborator on my repo to make it easier to push changes directly in the future.

tram-nguyen-n commented:
Hi Martin, I just pushed some updates to the pull request which includes a second iteration of the `clinvar_plot()` function where I tried to incorporate all of your feedback. Details are below and I also included a couple questions that I would appreciate your thoughts on. Thank you!

tram-nguyen-n commented:
- 	The Clinvar data is now a table in the database and is accessible to the user with the function `clinvar_data()`. This returns a table, exactly as it appears from the supplemental .csv of the AlphaMissense publication.

       - By default, the raw table currently loads the “label” field as a `dbl` but should probably be a factor. Should I make this change before importing the raw data for the user?

- 	`inst/scripts/import_clinvar_data.R` now only describes how the raw Clinvar data was imported from the AlphaMissense publication. It now uses an R solution for unzipping. However, both `download.file()` and `httr::GET()` failed since the URL requires authentication and is a 403 code status.  

 	- Using `lynx` through the terminal was the only solution that worked for me. Do you have suggestions for a better way to approach this?
 	
-	Munging steps of the Clinvar data for the purposes of integrating with AM data is now included in the helper function `clinvar_filter_cv_table`
-	`clinvar_plot` and other function names now match the prefix scheme of the package. 
     - Would you rather we changed it to `cv_plot` instead of `clinvar_plot` to match `af_` and `am_` ?
-	Documentation for the plotting and clinvar data are within one single `ClinVar.Rd`help page
-	Removed `LazyLoad: true` since we no longer have in-house data
-	Syntax fixes

mtmorgan commented on R/clinvar_plot.R:
```diff
@@ -0,0 +1,397 @@
+
+    ## Check if table is empty after filtering
+    ## This will work for a tibble or a data.frame
+    if (!nrow(alphamissense_table)) {
```
use `NROW()` (here and elsewhere) as best practice?

mtmorgan commented on R/clinvar_plot.R:
```diff
@@ -0,0 +1,397 @@
+
+    clinvar <- clinvar_data()
+
+    # Separate UniProt ID and Protein Variant
```
The indentation here suggests that this is outside the `if (missing(cv_table)) {...` but the logic puts it inside...

mtmorgan commented on R/clinvar_plot.R:
```diff
@@ -0,0 +1,397 @@
+            max = max(.data$am_pathogenicity, na.rm=TRUE),
+            min = min(.data$am_pathogenicity, na.rm=TRUE)
+        ) |>
+        ungroup()
```
I'd add a line `combined_data` so it is clear what is being returned (and so the return value is displayed if the function is ever invoked interactively, e.g., during debugging, since `f = function() { x <- 1 }; f()` returns `1`, but invisibly)

mtmorgan commented on R/clinvar_plot.R:
```diff
@@ -0,0 +1,397 @@
+#' @param alphamissense_table a table containing AlphaMissense
+#'    predictions for protein variants. By default, the table is
+#'    derived from `am_data("aa_substitution")`. Alternatively, a
+#'    user-defined \code{\link{tibble}} or \code{\link{data.frame}}
```
I think these types of links have roxygen markdown equivalents, <https://cran.r-project.org/web/packages/roxygen2/vignettes/rd-formatting.html#function-links>

mtmorgan commented on inst/scripts/import_clinvar_data.R:
```diff
@@ -0,0 +1,19 @@
+#lynx -accept_all_cookies=TRUE https://www.science.org/doi/suppl/10.1126/science.adg7492/suppl_file/science.adg7492_data_s1_to_s9.zip
```
`browseURL('...')` will open the browser and consequently download the file (to `~/Downloads` or wherever..).

Also, I'm not sure where I stand on the following and I'll let you make up your own mind on this... The problem you've pointed out is that accessing the data actually requires a subscription, so in that sense is it actually available for us to redistribute? If not, then (a) we might include a small subset that can be used in the package for examples / etc, maybe used with `clinvar_data_example()` and (b) provide instructions for the user to download the data to some location on disk (e.g., using `browseURL()`), and then `clinvar_data("~/Downloads/science.adg7492_data_s1_to_s9.zip")` would take that file and stick it in the database for subsequent use as `clinvar_data()`.

tram-nguyen-n replied:
Taking another look, the journal is subscription based but it seems the AM authors have actually made their article and all supplemental materials "Free Access"! My feelings are that it should be fine for us to include. The `browseURL(...)` works great, thank you for the suggestion. I'll go ahead with this approach instead of `lynx`.

mtmorgan commented on man/ClinVar.Rd:
I think the file should be named the same as the `R/` file, so `clinvar_plot.Rd`. This makes it easy to know where to look for the documentation.

mtmorgan commented on tests/testthat/test-clinvar_plot.R:
```diff
@@ -0,0 +1,166 @@
+    expect_equal(res |> nrow(), 9557L)
+
+    ## Test case when am_table is provided by user
+    am_data <- data.frame(uniprot_id = c("A", "A", "A", "A", "A", "B"),
```
It would be great to format the unit tests in the Martin Morgan style, along the lines of
```r
    am_data <- data.frame(
        uniprot_id = c("A", "A", "A", "A", "A", "B"),
        ...
    )
...
```
similar formatting for `expect_error()` and `expect_identical()` phrases below.

mtmorgan commented on tests/testthat/test-clinvar_plot.R:
```diff
@@ -0,0 +1,166 @@
+    ## Test case when bad am_table is given i.e uniprotID with no hits
+    expect_error(
+        clinvar_filter_am_table(uID = "C", am_table = am_data),
+        paste0(
```
trivial, but `paste(...)` with no trailing spaces in the arguments would accomplish the same, and is consistent with `stop(...)` and `message(...)`.

mtmorgan reviewed (commented):
This looks really great Tram!

There's one substantive comment on the inst/scripts file about whether it is appropriate for us to redistribute the supplemental data (it's behind a paywall, so not freely available...).

For your questions

>        - By default, the raw table currently loads the “label” field as a `dbl` but should probably be a factor. Should I make this change before importing the raw data for the user?

Yes, I think it should be made as the data is being imported.

>  	- Using `lynx` through the terminal was the only solution that worked for me. Do you have suggestions for a better way to approach this?

I think `browseURL()` on the link to the zip opens the browser and places the file in the browser 'Downloads' folder.

>      - Would you rather we changed it to `cv_plot` instead of `clinvar_plot` to match `af_` and `am_` ?

I think I'm going to be terribly inconsistent and say that I like them named `clinvar_plot` etc.

mtmorgan commented:
> > > ```
> > >    - By default, the raw table currently loads the “label” field as a `dbl` but should probably be a factor. Should I make this change before importing the raw data for the user?
> > > ```
> > 
> > 
> > Yes, I think it should be made as the data is being imported.
> 
> Hi Martin, how should I implement this change during the data import? Do I need to modify the `sql_template` for `am_data_import_csv()`?

Yes maybe make `template = "import_csv"` an argument to `am_data_import_csv()`, create inst/sql/import_clinvar_csv.sql,  and invoke with `template = "import_clinvar_csv"` from `clinvar_data()`?

mtmorgan commented:
I actually meant recode as 0: benign, 1: pathogenic so that the user didn't have to know this detail. If you're OK with that change, I'll implement it along the lines of the following, and then accept your PR.

Just to outline how I worked through the interesting puzzle of creating a factor and recoding... To explore I used something like

```r
con <- dbConnect(duckdb())
sql <- "..."
DBI::dbGetQuery(con, sql)
```
which is more interactive so easier to try things out.

## Enum types

DuckDB has the [enum data type](https://duckdb.org/docs/sql/data_types/enum.html), and an enum is understood by dbplyr to be a factor. Here's the R code pieced together from the enum link

```r
con <- dbConnect(duckdb())
sql <- "
CREATE TYPE mood AS ENUM ('sad', 'ok', 'happy');
CREATE TABLE person (name TEXT, current_mood mood);
INSERT INTO person
VALUES ('Pedro', 'happy'), ('Mark', NULL), ('Pagliacci', 'sad'), ('Mr. Mackey', 'ok');
"
DBI::dbExecute(con, sql)
```
and proof that this is represented as a factor
```r
dplyr::tbl(con, "person")
## # Source:   table<person> [4 x 2]
## # Database: DuckDB v0.10.2 [ma38727@Darwin 23.5.0:R 4.4.1/:memory:]
##   name       current_mood
##   <chr>      <fct>       
## 1 Pedro      happy       
## 2 Mark       NA          
## 3 Pagliacci  sad         
## 4 Mr. Mackey ok          
```

## Recode on select

```sql
SELECT label, * EXCLUDE(label)
FROM ...
```
actually selects `label` (for special treatment) and then everything else. So
```sql
SELECT 
    CASE
        WHEN label = 0 THEN 'benign'
        WHEN label = 1 THEN 'pathogenic'
    END AS label,
    * EXCLUDE(label)
FROM ...
```
gets label as character and everything else automatically.

Putting these together the best I could do was to create 'label' as text and then coerce to enum. I have (the value of `sql` would go in `import_clinvar_csv.sql`)
```r
con <- dbConnect(duckdb())
sql <- "
-- create levels for the label column
CREATE TYPE label_levels AS ENUM ('benign', 'pathogenic');
-- create the clinvar table, but with label as TEXT
CREATE TABLE clinvar AS
SELECT 
    CASE
        WHEN label = 0 THEN 'benign'
        WHEN label = 1 THEN 'pathogenic'
    END AS label,
    * EXCLUDE(label)
FROM read_csv_auto(
    'inst/extdata/science.adg7492_data_s5.csv.gz', 
    delim = ',', 
    compression = 'gzip'
);
-- coerce label to factor
ALTER TABLE clinvar ALTER label TYPE label_levels;
"
DBI::dbExecute(con, sql)
```
I then have a 'label' column that is a factor
```r
tbl(con, "clinvar") |> dplyr::count(label)
## # Source:   SQL [2 x 2]
## # Database: DuckDB v0.10.2 [ma38727@Darwin 23.5.0:R 4.4.1/:memory:]
##   label          n
##   <fct>      <dbl>
## 1 benign     51988
## 2 pathogenic 30884
```

tram-nguyen-n commented:
Ah neat, thank you for laying this out! I wasn't aware of the `enum` type.

And yes, totally OK with the recode 0: benign, 1: pathogenic. Thanks in advance for implementing these changes! I've also edited the `clinvar_plot()` function and man page for `clinvar_data()` so that it will be compatible with the recode once implemented.

mtmorgan commented:
I think the unit tests are failing for reasons not related to the database import?

```r
> devtools::test(filter = "clinvar")
ℹ Testing AlphaMissenseR
✔ | F W  S  OK | Context
⠏ |          0 | clinvar_plot                                                   * [12:36:30][info] 'alphamissense_table' not provided, using default 'am_data("aa_substitution")' table accessed through the AlphaMissenseR package
⠋ |          1 | clinvar_plot                                                   * [12:36:33][info] 'clinvar_table' not provided, using default ClinVar dataset in AlphaMissenseR package
✖ | 8       20 | clinvar_plot [6.4s]
────────────────────────────────────────────────────────────────────────────────
Error (test-clinvar_plot.R:37:5): clinvar_filter_cv_table() works
<dplyr:::mutate_error/rlang_error/error/condition>
Error in `mutate(tidyr::separate(as_tibble(clinvar), .data$protein_variant,
    into = c("uniprot_id", "protein_variant"), sep = ":"), cv_class = as.factor(.data$label_category))`: i In argument: `cv_class = as.factor(.data$label_category)`.
Caused by error in `.data$label_category`:
! Column `label_category` not found in `.data`.

Failure (test-clinvar_plot.R:103:5): clinvar_prepare_data_for_plot() works
`object` (`actual`) not identical to `expected` (`expected`).

`actual$protein_variant`:   "R3L" "K4L" "F50G" "M1A" "M2C" "G5L"
`expected$protein_variant`: "R3L" "K4L" "F50G" "M2C" "G5L" "M1A"

  `actual$am_pathogenicity`: 0.5 0.2 0.6 0.2 0.8 0.4
`expected$am_pathogenicity`: 0.5 0.2 0.6 0.8 0.4 0.2

    actual$am_class | expected$am_class
[1] "ambiguous"     | "ambiguous"       [1]
[2] "benign"        | "benign"          [2]
[3] "pathogenic"    | "pathogenic"      [3]
[4] "benign"        - "pathogenic"      [4]
[5] "pathogenic"    - "ambiguous"       [5]
[6] "ambiguous"     - "benign"          [6]

  `actual$aa_pos`: 3 4 50 1 2 5
`expected$aa_pos`: 3 4 50 2 5 1

  `actual$cv_class`: NA NA NA 1 0 0
`expected$cv_class`: NA NA NA 0 0 1

    levels(actual$code_color) | levels(expected$code_color)
[1] "AM ambiguous"            | "AM ambiguous"              [1]
[2] "AM benign"               | "AM benign"                 [2]
[3] "AM pathogenic"           | "AM pathogenic"             [3]
                              - "CV benign"                 [4]
                              - "CV pathogenic"             [5]

    actual$code_color | expected$code_color
[1] "AM ambiguous "   | "AM ambiguous "     [1]
[2] "AM benign    "   | "AM benign    "     [2]
[3] "AM pathogenic"   | "AM pathogenic"     [3]
[4] "NA           "   - "CV benign    "     [4]
[5] "NA           "   - "CV benign    "     [5]
[6] "NA           "   - "CV pathogenic"     [6]

  `actual$max`: 0.5 0.2 0.8 0.2 0.8 0.5
`expected$max`: 0.5 0.2 0.8 0.8 0.5 0.2

  `actual$min`: 0.4 0.2 0.6 0.2 0.6 0.4
`expected$min`: 0.4 0.2 0.6 0.6 0.4 0.2

Failure (test-clinvar_plot.R:119:5): clinvar_prepare_data_for_plot() works
as.vector(pull(select(object, code_color))) (`actual`) not identical to `color_names` (`expected`).

    actual          | expected
[1] "AM ambiguous"  | "AM ambiguous"  [1]
[2] "AM benign"     | "AM benign"     [2]
[3] "AM pathogenic" | "AM pathogenic" [3]
[4] NA              - "CV benign"     [4]
[5] NA              - "CV benign"     [5]
[6] NA              - "CV pathogenic" [6]

Failure (test-clinvar_plot.R:185:5): 'clinvar_plot()' works
ggplot_build(plot)$data[[1]]$stroke (`actual`) not identical to `stroke_vec` (`expected`).

  `actual`: 0 0 NA NA NA
`expected`: 0 0  2  2  2

Failure (test-clinvar_plot.R:188:5): 'clinvar_plot()' works
ggplot_build(plot)$data[[1]]$shape (`actual`) not identical to `shape_vec` (`expected`).

  `actual`: 19 19 NA NA NA
`expected`: 19 19 21 21 21

Failure (test-clinvar_plot.R:191:5): 'clinvar_plot()' works
ggplot_build(plot)$data[[1]]$size (`actual`) not identical to `size_vec` (`expected`).

  `actual`: 2 2 NA NA NA
`expected`: 2 2  4  4  4

Failure (test-clinvar_plot.R:194:5): 'clinvar_plot()' works
ggplot_build(plot)$data[[1]]$colour (`actual`) not identical to `color_vec` (`expected`).

`actual`:   "gray" "#89d5f5" "grey50" "grey50" "grey50"
`expected`: "gray" "#89d5f5" "black"  "black"  "black"

Failure (test-clinvar_plot.R:197:5): 'clinvar_plot()' works
ggplot_build(plot)$data[[1]]$fill (`actual`) not identical to `fill_vec` (`expected`).

`actual`:   "gray" "#89d5f5" "grey50"  "grey50"  "grey50"
`expected`: "gray" "#89d5f5" "#007cb0" "#007cb0" "#c70606"
────────────────────────────────────────────────────────────────────────────────

══ Results ═════════════════════════════════════════════════════════════════════
Duration: 6.6 s

── Failed tests ────────────────────────────────────────────────────────────────
Error (test-clinvar_plot.R:37:5): clinvar_filter_cv_table() works
<dplyr:::mutate_error/rlang_error/error/condition>
Error in `mutate(tidyr::separate(as_tibble(clinvar), .data$protein_variant,
    into = c("uniprot_id", "protein_variant"), sep = ":"), cv_class = as.factor(.data$label_category))`: i In argument: `cv_class = as.factor(.data$label_category)`.
Caused by error in `.data$label_category`:
! Column `label_category` not found in `.data`.

Failure (test-clinvar_plot.R:103:5): clinvar_prepare_data_for_plot() works
`object` (`actual`) not identical to `expected` (`expected`).

`actual$protein_variant`:   "R3L" "K4L" "F50G" "M1A" "M2C" "G5L"
`expected$protein_variant`: "R3L" "K4L" "F50G" "M2C" "G5L" "M1A"

  `actual$am_pathogenicity`: 0.5 0.2 0.6 0.2 0.8 0.4
`expected$am_pathogenicity`: 0.5 0.2 0.6 0.8 0.4 0.2

    actual$am_class | expected$am_class
[1] "ambiguous"     | "ambiguous"       [1]
[2] "benign"        | "benign"          [2]
[3] "pathogenic"    | "pathogenic"      [3]
[4] "benign"        - "pathogenic"      [4]
[5] "pathogenic"    - "ambiguous"       [5]
[6] "ambiguous"     - "benign"          [6]

  `actual$aa_pos`: 3 4 50 1 2 5
`expected$aa_pos`: 3 4 50 2 5 1

  `actual$cv_class`: NA NA NA 1 0 0
`expected$cv_class`: NA NA NA 0 0 1

    levels(actual$code_color) | levels(expected$code_color)
[1] "AM ambiguous"            | "AM ambiguous"              [1]
[2] "AM benign"               | "AM benign"                 [2]
[3] "AM pathogenic"           | "AM pathogenic"             [3]
                              - "CV benign"                 [4]
                              - "CV pathogenic"             [5]

    actual$code_color | expected$code_color
[1] "AM ambiguous "   | "AM ambiguous "     [1]
[2] "AM benign    "   | "AM benign    "     [2]
[3] "AM pathogenic"   | "AM pathogenic"     [3]
[4] "NA           "   - "CV benign    "     [4]
[5] "NA           "   - "CV benign    "     [5]
[6] "NA           "   - "CV pathogenic"     [6]

  `actual$max`: 0.5 0.2 0.8 0.2 0.8 0.5
`expected$max`: 0.5 0.2 0.8 0.8 0.5 0.2

  `actual$min`: 0.4 0.2 0.6 0.2 0.6 0.4
`expected$min`: 0.4 0.2 0.6 0.6 0.4 0.2

Failure (test-clinvar_plot.R:119:5): clinvar_prepare_data_for_plot() works
as.vector(pull(select(object, code_color))) (`actual`) not identical to `color_names` (`expected`).

    actual          | expected
[1] "AM ambiguous"  | "AM ambiguous"  [1]
[2] "AM benign"     | "AM benign"     [2]
[3] "AM pathogenic" | "AM pathogenic" [3]
[4] NA              - "CV benign"     [4]
[5] NA              - "CV benign"     [5]
[6] NA              - "CV pathogenic" [6]

Failure (test-clinvar_plot.R:185:5): 'clinvar_plot()' works
ggplot_build(plot)$data[[1]]$stroke (`actual`) not identical to `stroke_vec` (`expected`).

  `actual`: 0 0 NA NA NA
`expected`: 0 0  2  2  2

Failure (test-clinvar_plot.R:188:5): 'clinvar_plot()' works
ggplot_build(plot)$data[[1]]$shape (`actual`) not identical to `shape_vec` (`expected`).

  `actual`: 19 19 NA NA NA
`expected`: 19 19 21 21 21

Failure (test-clinvar_plot.R:191:5): 'clinvar_plot()' works
ggplot_build(plot)$data[[1]]$size (`actual`) not identical to `size_vec` (`expected`).

  `actual`: 2 2 NA NA NA
`expected`: 2 2  4  4  4

Failure (test-clinvar_plot.R:194:5): 'clinvar_plot()' works
ggplot_build(plot)$data[[1]]$colour (`actual`) not identical to `color_vec` (`expected`).

`actual`:   "gray" "#89d5f5" "grey50" "grey50" "grey50"
`expected`: "gray" "#89d5f5" "black"  "black"  "black"

Failure (test-clinvar_plot.R:197:5): 'clinvar_plot()' works
ggplot_build(plot)$data[[1]]$fill (`actual`) not identical to `fill_vec` (`expected`).

`actual`:   "gray" "#89d5f5" "grey50"  "grey50"  "grey50"
`expected`: "gray" "#89d5f5" "#007cb0" "#007cb0" "#c70606"

[ FAIL 8 | WARN 0 | SKIP 0 | PASS 20 ]

Don't worry, you'll get it.
```

tram-nguyen-n commented:
Oops sorry, forgot to change one instance of the `label_category` back to `label`. Would you mind trying again please?

mtmorgan commented:
Thanks @tram-nguyen-n for an amazing contribution! I implemented my own updates to the unit tests, so the PR is marked as 'Closed' but really it is merged. 

That was a really great effort and I hope not too painful for you! I changed your role from 'ctb' to 'aut' to reflect the awesome contribution.

tram-nguyen-n commented:
Thanks @mtmorgan for this wonderful collaboration and for your consistent help along the way! It was a great experience and I learned a ton working with the package!
